File: rust/src/models/coupon_event.rs
// this file is @generated
use serde::{Deserialize, Serialize};

use super::{coupon_event_data::CouponEventData, event_id::EventId, event_type::EventType};

#[derive(Clone, Debug, Default, PartialEq, Deserialize, Serialize)]
pub struct CouponEvent {
    #[serde(flatten)]
    pub flatten_couponeventdata: CouponEventData,

    pub id: EventId,

    pub timestamp: String,

    pub r#type: EventType,
}

impl CouponEvent {
    pub fn new(id: EventId, timestamp: String, r#type: EventType) -> Self {
        Self {
            flatten_couponeventdata: Default::default(),
            id,
            timestamp,
            r#type,
        }
    }
}

File: typescript/src/models/couponEvent.ts
// this file is @generated
import { parseDateTime } from "../datetime";
import { type CouponEventData, CouponEventDataSerializer } from "./couponEventData";
import { type EventId, EventIdSerializer } from "./eventId";
import { type EventType, EventTypeSerializer } from "./eventType";

export interface CouponEvent {
  flattenCouponeventdata: CouponEventData;

  id: EventId;

  timestamp: Date;

  type: EventType;
}

export const CouponEventSerializer = {
  _fromJsonObject(object: any): CouponEvent {
    return {
      flattenCouponeventdata: CouponEventDataSerializer._fromJsonObject(
        object["__flatten_couponeventdata"]
      ),
      id: EventIdSerializer._fromJsonObject(object["id"]),
      timestamp: parseDateTime(object["timestamp"]),
      type: EventTypeSerializer._fromJsonObject(object["type"]),
    };
  },

  _toJsonObject(self: CouponEvent): any {
    return {
      __flatten_couponeventdata: CouponEventDataSerializer._toJsonObject(
        self.flattenCouponeventdata
      ),
      id: EventIdSerializer._toJsonObject(self.id),
      timestamp: self.timestamp,
      type: EventTypeSerializer._toJsonObject(self.type),
    };
  },
};
